src/test/ui/const-generics/min_const_generics/macro.rs

```diff
@@ -0,0 +1,30 @@
+#![feature(min_const_generics)]
+
+struct Example<const N: usize>;
+
+macro_rules! external_macro {
+  () => {{
+    //~^ ERROR expected type
+    const X: usize = 1337;
+    X
+  }}
+}
+
+fn main() {
+  let _ok = Example::<{
+    #[macro_export]
+    macro_rules! gimme_a_const {
+      ($rusty: ident) => {{
+        let $rusty = 3;
+        $rusty
+      }}
+    };
+    gimme_a_const!(run)
+  }>;
+
+  let _ok = Example::<{ external_macro!() }>;
+
+  let _fail = Example::<external_macro!()>;
```

Comment:
It's interesting that this fails :thinking: cc @petrochenkov @varkor what would be needed to get this to work?
Might be good to open an separate issue for this.

Would this also need an `GenericArg::Ambiguous` variant in the AST?

Reply:
`external_macro!()` is not a literal or block expression, so it's being parsed as a type argument. We don't want this to work.

Reply:
but if the macro itself returns a block, it doesn't seem that that gets parsed as a const. It's fine if this is the intended behavior, just a bit surprising